### PR TITLE
IIO/Maxim fixes

### PR DESCRIPTION
**Pull request by CiprianRegus**
## Pull Request Description

Various fixes for iio and maxim platform drivers.

## PR Type
- [x] Bug fix (change that fixes an issue)
- [ ] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [x] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

**Review by amiclaus — APPROVED** [review]
lgtm

File: drivers/platform/maxim/max32650/maxim_gpio_irq.c [hunk 1]
```diff
@@ -106,6 +106,11 @@ void GPIO2_IRQHandler()
 }
 #endif
 
+void GPIO3_IRQHandler()
```
Comment (buha) [inline drivers/platform/maxim/max32650/maxim_gpio_irq.c:109]: needs `#ifdef MXC_GPIO3`

Comment (CiprianRegus) [reply drivers/platform/maxim/max32650/maxim_gpio_irq.c:109]: not really, the max32650 has 4 GPIO ports. In fact, the other `ifdefs` should be removed.

Comment (buha) [reply drivers/platform/maxim/max32650/maxim_gpio_irq.c:109]: well, i picked a random one to comment on, but i think you got the point

Comment (buha) [reply drivers/platform/maxim/max32650/maxim_gpio_irq.c:109]: it seems to me easier to maintain if you copy paste code that covers the GPIO0 to GPIO3 range (or beyond) in all maxim_gpio_irq.c files without counting how many of them does the microcontroller really have

**Review by rbolboac — APPROVED** [review]

**Pull request merged**